src/interface/src/app/treatments/treatment-project-area/treatment-project-area.component.ts

```diff
@@ -34,6 +38,35 @@ export class TreatmentProjectAreaComponent implements OnDestroy {
   ) {}
 
   opacity = this.mapConfigState.treatedStandsOpacity$;
+  summary$ = this.treatmentsState.summary$;
+  projectAreaId$ = this.treatmentsState.projectAreaId$;
+
+  activeProjectArea$: Observable<TreatmentProjectArea | undefined> =
+    combineLatest([
+      this.summary$,
+      this.projectAreaId$?.pipe(distinctUntilChanged()),
+    ]).pipe(
+      map(([summary, projectAreaId]) => {
+        return summary?.project_areas.find(
+          (p) => p.project_area_id === projectAreaId
+        );
+      })
+    );
+
+  treatedStands$ = this.activeProjectArea$?.pipe(
+    map(
+      (area) =>
+        area?.prescriptions?.reduce(
```

Comment:
This is the same code used on src/interface/src/app/treatments/map-project-areas/map-project-areas.component.ts , can you extract this to a function or method so we don't repeat this code?